deployment/amazon-marketing-cloud-uploader-from-aws.yaml

```diff
@@ -664,6 +665,42 @@ Resources:
         CustomerManagedKey: !Ref CustomerManagedKey
 
   # API stack
+
+  AmcApiAccessRole:
+    Metadata:
+      cfn_nag:
+        rules_to_suppress:
+          - id: W28
+            reason: "Role name is constructed with stack name. Updates will keep the existing role name."
+    Type: AWS::IAM::Role
+    Properties:
+      Description: "IAM role to grant access to the AMC API."
+      RoleName: !Join [ "-", [ { "Ref": "AWS::StackName" }, "AmcApiAccessRole" ] ]
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !GetAtt ApiStack.Outputs.ApiHandlerRoleArn
+            Action:
+              - sts:AssumeRole
+          - Effect: Allow
+            Principal:
+              AWS: !GetAtt AmcUploadLambdaExecutionRole.Arn
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: AmcApiAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Sid: "AmcEndpointAccessPolicy"
+                Effect: Deny
```

Comment:
Is this by design to deny access to AMC?

Reply:
yes.